fe1-web/src/features/wallet/screens/WalletHome.tsx

```diff
@@ -1,49 +1,117 @@
+import { RollCallToken } from '../objects/RollCallToken';
 
 const styles = StyleSheet.create({
+  homeContainer: {
+    ...containerStyles.centeredXY,
+    padding: 30,
+  } as ViewStyle,
   smallPadding: {
     padding: '1rem',
   } as ViewStyle,
-  largePadding: {
-    padding: '2rem',
+  tokenSelectContainer: {
+    flexDirection: 'row',
+    marginTop: 30,
+    marginBottom: 20,
   } as ViewStyle,
 });
 
+const rollCallSelector = makeEventByTypeSelector<RollCall>(LaoEventType.ROLL_CALL);
+
 /**
- * Wallet home screen
+ * Wallet UI once the wallet is synced
  */
 const WalletHome = () => {
-  // FIXME: Navigation should use a defined type here (instead of any)
+  const [tokens, setTokens] = useState<RollCallToken[]>();
+  const [selectedTokenIndex, setSelectedTokenIndex] = useState(-1);
+  const [isDebug, setIsDebug] = useState(false);
+  const rollCalls = useSelector(rollCallSelector);
+  const lao = useSelector(selectCurrentLao);
   const navigation = useNavigation<any>();
 
-  function importSeed() {
-    if (WalletStore.hasSeed()) {
-      navigation.navigate(STRINGS.navigation_synced_wallet);
+  useEffect(() => {
+    if (lao && rollCalls[lao.id.valueOf()]) {
+      Wallet.recoverWalletRollCallTokens(rollCalls, lao)
+        .then((rct) => {
+          if (rct.length > 0) {
+            setTokens(rct);
+            setSelectedTokenIndex(0);
+          }
+        })
+        .catch((e) => {
+          console.debug(e);
+        });
+    } else {
+      // Clear tokens screen state
+      setSelectedTokenIndex(-1);
+      setTokens(undefined);
+    }
+  }, [rollCalls, isDebug, lao]);
+
+  const toggleDebugMode = () => {
+    if (isDebug) {
+      clearDummyWalletState();
+      setIsDebug(false);
     } else {
-      navigation.navigate(STRINGS.navigation_insert_seed_tab_wallet);
+      createDummyWalletState().then(() => setIsDebug(true));
+    }
+  };
+  const tokenInfos = () => {
+    if (selectedTokenIndex !== -1 && tokens) {
+      const rollCallName = `Roll Call name: ${tokens[selectedTokenIndex].rollCallName.valueOf()}`;
+      return (
+        <View style={containerStyles.centeredXY}>
+          <TextBlock size={18} text={rollCallName} />
+          <QRCode value={tokens[selectedTokenIndex].token.publicKey.valueOf()} visibility />
+        </View>
+      );
     }
-  }
+    return <TextBlock text={STRINGS.no_tokens_in_wallet} />;
+  };
 
   return (
-    <View style={containerStyles.centeredY}>
-      <TextBlock bold text={STRINGS.welcome_to_wallet_display} />
+    <View style={styles.homeContainer}>
+      <TextBlock bold text={STRINGS.wallet_welcome} />
+      <View style={styles.tokenSelectContainer}>
+        {tokens && (
+          <RollCallTokensDropDown
+            rollCallTokens={tokens}
+            onIndexChange={setSelectedTokenIndex}
+            selectedTokenIndex={selectedTokenIndex}
+          />
+        )}
+      </View>
+      {tokens && tokenInfos()}
```

Comment:
I would either inline the function or make it a component, having a non-component returning jsx is a bit counterintuitive to me? 😅 But I guess this is a design choice

Reply:
Why would that not be intuitive ?

Reply:
A component is nothing else than a function returning jsx so I usually expect non-component functions (non-captialized) not to return jsx :)